File: app/src/main/java/vn/loitp/app/activity/customviews/actionbar/_lib/LActionBar.java
package vn.loitp.app.activity.customviews.actionbar._lib;

import android.content.Context;
import android.util.AttributeSet;
import android.view.Gravity;
import android.view.View;
import android.widget.ImageView;
import android.widget.RelativeLayout;
import android.widget.TextView;

import com.daimajia.androidanimations.library.Techniques;
import com.github.mmin18.widget.RealtimeBlurView;

import vn.loitp.app.utilities.LAnimationUtil;
import vn.loitp.livestar.R;

/**
 * Created by www.muathu@gmail.com on 5/13/2017.
 */

public class LActionBar extends RelativeLayout {
    private final String TAG = getClass().getSimpleName();
    private ImageView ivIconBack;
    private ImageView ivIconMenu;
    private TextView tvTitle;
    private RealtimeBlurView realtimeBlurView;
    private View shadowView;

    public ImageView getIvIconBack() {
        return ivIconBack;
    }

    public ImageView getIvIconMenu() {
        return ivIconMenu;
    }

    public TextView getTvTitle() {
        return tvTitle;
    }

    public LActionBar(Context context, AttributeSet attrs) {
        super(context, attrs);
        init();
    }

    public LActionBar(Context context, AttributeSet attrs, int defStyle) {
        super(context, attrs, defStyle);
        init();
    }

    private void init() {
        inflate(getContext(), R.layout.view_l_action_bar, this);

        this.ivIconBack = (ImageView) findViewById(R.id.iv_icon_back);
        this.ivIconMenu = (ImageView) findViewById(R.id.iv_icon_menu);
        this.tvTitle = (TextView) findViewById(R.id.tv_title);
        this.realtimeBlurView = (RealtimeBlurView) findViewById(R.id.blur_view);
        this.shadowView = (View) findViewById(R.id.shadow_view);

        ivIconBack.setOnClickListener(new OnClickListener() {
            @Override
            public void onClick(View v) {
                LAnimationUtil.play(v, Techniques.Pulse);
                if (callback != null) {
                    callback.onClickBack();
                }
            }
        });
        ivIconMenu.setOnClickListener(new OnClickListener() {
            @Override
            public void onClick(View v) {
                LAnimationUtil.play(v, Techniques.Pulse);
                if (callback != null) {
                    callback.onClickMenu();
                }
            }
        });
    }

    public interface Callback {
        public void onClickBack();

        public void onClickMenu();
    }

    private Callback callback;

    public void setOnClickBack(Callback onClickBack) {
        this.callback = onClickBack;
    }

    public void setTvTitle(String title) {
        tvTitle.setText(title);
    }

    public void hideBackIcon() {
        ivIconBack.setVisibility(GONE);
    }

    public void inviBackIcon() {
        ivIconBack.setVisibility(INVISIBLE);
    }

    public void hideMenuIcon() {
        ivIconMenu.setVisibility(GONE);
    }

    public void showMenuIcon() {
        ivIconMenu.setVisibility(VISIBLE);
    }

    public void setTvTitlePositionLeft() {
        tvTitle.setGravity(Gravity.LEFT | Gravity.CENTER_VERTICAL);
    }

    public void setTvTitlePositionCenter() {
        tvTitle.setGravity(Gravity.CENTER);
    }

    public void setImageMenuIcon(int drawableRes) {
        ivIconMenu.setImageResource(drawableRes);
    }

    public void setImageBackIcon(int drawableRes) {
        ivIconBack.setImageResource(drawableRes);
    }

    public void hideBlurView() {
        realtimeBlurView.setVisibility(GONE);
    }

    public void hideShadowView() {
        shadowView.setVisibility(GONE);
    }
}

File: core/src/main/java/vn/loitp/views/actionbar/LActionBar.kt
package vn.loitp.views.actionbar

import android.content.Context
import android.util.AttributeSet
import android.view.Gravity
import android.view.View
import android.widget.ImageView
import android.widget.RelativeLayout
import android.widget.TextView
import com.core.utilities.LAnimationUtil
import com.daimajia.androidanimations.library.Techniques
import loitp.core.R
import vn.loitp.views.realtimeblurview.RealtimeBlurView

/**
 * Created by www.muathu@gmail.com on 5/13/2017.
 */

class LActionBar : RelativeLayout {
    private val TAG = javaClass.simpleName
    var ivIconBack: ImageView? = null
    var ivIconMenu: ImageView? = null
    var tvTitle: TextView? = null
    private var realtimeBlurView: RealtimeBlurView? = null
    private var shadowView: View? = null
    private var callback: Callback? = null

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {
        init()
    }

    constructor(context: Context, attrs: AttributeSet, defStyle: Int) : super(context, attrs, defStyle) {
        init()
    }

    private fun init() {
        View.inflate(context, R.layout.view_l_action_bar, this)

        this.ivIconBack = findViewById<ImageView>(R.id.iv_icon_back)
        this.ivIconMenu = findViewById<ImageView>(R.id.iv_icon_menu)
        this.tvTitle = findViewById<TextView>(R.id.tv_title)
        this.realtimeBlurView = findViewById<RealtimeBlurView>(R.id.blur_view)
        this.shadowView = findViewById<View>(R.id.shadow_view)

        ivIconBack!!.setOnClickListener { v ->
            LAnimationUtil.play(v, Techniques.Pulse)
            if (callback != null) {
                callback!!.onClickBack()
            }
        }
        ivIconMenu!!.setOnClickListener { v ->
            LAnimationUtil.play(v, Techniques.Pulse)
            if (callback != null) {
                callback!!.onClickMenu()
            }
        }
    }

    interface Callback {
        fun onClickBack()

        fun onClickMenu()
    }

    fun setOnClickBack(onClickBack: Callback) {
        this.callback = onClickBack
    }

    fun setTvTitle(title: String) {
        tvTitle!!.text = title
    }

    fun hideBackIcon() {
        ivIconBack!!.visibility = View.GONE
    }

    fun inviBackIcon() {
        ivIconBack!!.visibility = View.INVISIBLE
    }

    fun hideMenuIcon() {
        ivIconMenu!!.visibility = View.GONE
    }

    fun showMenuIcon() {
        ivIconMenu!!.visibility = View.VISIBLE
    }

    fun setTvTitlePositionLeft() {
        tvTitle!!.gravity = Gravity.START or Gravity.CENTER_VERTICAL
    }

    fun setTvTitlePositionCenter() {
        tvTitle!!.gravity = Gravity.CENTER
    }

    fun setImageMenuIcon(drawableRes: Int) {
        ivIconMenu!!.setImageResource(drawableRes)
    }

    fun setImageBackIcon(drawableRes: Int) {
        ivIconBack!!.setImageResource(drawableRes)
    }

    fun hideBlurView() {
        realtimeBlurView!!.visibility = View.GONE
    }

    fun hideShadowView() {
        shadowView!!.visibility = View.GONE
    }
}